app/models/student.rb

```diff
@@ -56,6 +56,44 @@ def valid_forms(type)
     end
   end
 
+  def valid_forms_for_year(type, year)
+    case type
+    when :team_dues
+      team_dues.valid_for(year)
+    when :media_consents
+      media_consents.valid_for(year)
+    when :stims
+      stims.valid_for(year)
+    when :safety_tests
+      safety_tests.valid_for(year)
+    end
+  end
+
+  # A student will never have a permissions slip with
+  # the same type and trip as another permission slip
+  def find_permission_slip(type, trip_id)
+    permission_slips.where(type: type, trip_id: trip_id)
+  end
+
+  # A student will never have two payments for the same
+  # payment requirement
+  def find_payment(required_payment_id)
+    payments.where(required_payment_id: required_payment_id)
+  end
+
+  def eligible_for_trip(trip)
+    year = trip.end_date.year
```

Comment:
This won't work in some cases like preseason events. Imagine an event in November 2017 like Brunswick. The students should have forms for the 2018 year.